packages/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,624 @@
+import React, { Component, Fragment, ReactNode, RefObject } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { emotion } from '@leafygreen-ui/lib';
+
+const { css, cx } = emotion;
+
+const rootPopoverStyle = css`
+  transition: transform 150ms ease-in-out, opacity 150ms ease-in-out;
+  position: absolute;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+type RefPosition = {
+  top: number;
+  bottom: number;
+  left: number;
+  right: number;
+  height: number;
+  width: number;
+};
+
+type AbsolutePositionObject = {
+  top?: string | number;
+  bottom?: string | number;
+  left?: string | number;
+  right?: string | number;
+};
+
+const defaultRefPosition = {
+  top: 0,
+  bottom: 0,
+  left: 0,
+  right: 0,
+  height: 0,
+  width: 0,
+};
+
+type Alignment = 'top' | 'bottom' | 'left' | 'right';
+type Justify = 'start' | 'middle' | 'end';
+
+// We transform 'middle' into 'center-vertical' or 'center-horizontal' for internal use,
+// So both Justify and Justification are needed, where the same is not true for Alignment.
+type Justification =
+  | 'top'
+  | 'bottom'
+  | 'left'
+  | 'right'
+  | 'center-vertical'
+  | 'center-horizontal';
+
+type AbstractPosition = {
+  alignment?: Alignment;
+  justification?: Justification;
+};
+
+type DefaultProps = {
+  align: Alignment;
+  justify: Justify;
+  active: boolean;
+};
+
+interface Props {
+  children?: ReactNode;
+  active: boolean;
+  className?: string;
+  align: Alignment;
+  justify: Justify;
+  refEl?: RefObject<HTMLElement>;
+  withoutPortal?: boolean;
+  getUpdatePosition?: Function;
+}
+interface State {
+  windowHeight: number;
+  windowWidth: number;
+  hasMounted: boolean;
+  referenceElPos: RefPosition;
+  contentElPos: RefPosition;
+  referenceElement: HTMLElement | null;
+}
+
+export default class Popover extends Component<Props, State> {
+  static displayName = 'Popover';
+
+  static propTypes = {
+    children: PropTypes.node,
+    active: PropTypes.bool,
+    className: PropTypes.string,
+    align: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
```

Comment:
This should use a string enum and then `Object.keys(Align)`. Ditto below.

Reply:
👍